inductiva/fluids/scenarios/wind_terrain/wind_terrain.py

```diff
@@ -0,0 +1,282 @@
+            random.uniform(0, 1),
+            random.uniform(0, 1),
+            random.uniform(0, 1)
+        ]
+
+        x_grid, y_grid, z_elevation = inductiva.generative.generate_grid_elevation(
+            x_range, y_range, x_num, y_num, corner_values, initial_roughness,
+            roughness_factor)
+
+        terrain = pv.StructuredGrid(x_grid, y_grid, z_elevation * height_factor)
+
+        return cls(terrain)
+
+    def to_text_file(self, text_file_path: str):
+        """Saves the terrain to a text file.
+        
+        Args
+            text_file_path: Path to the text file
+        """
+
+        terrain_geometry = self.mesh.extract_geometry()
+        terrain_geometry.save(text_file_path)
+
+    @property
+    def bounds(self):
+        """Returns the bounds of the terrain."""
+
+        terrain_bounds = {
+            "x": [self.mesh.bounds[0], self.mesh.bounds[1]],
+            "y": [self.mesh.bounds[2], self.mesh.bounds[3]],
+            "z": [self.mesh.bounds[4], self.mesh.bounds[5]]
+        }
+
+        return terrain_bounds
+
+    @property
+    def center(self):
+        """Returns the center of the terrain."""
+
+        terrain_center = {
+            "x": (self.mesh.bounds[0] + self.mesh.bounds[1]) / 2,
+            "y": (self.mesh.bounds[2] + self.mesh.bounds[3]) / 2
+        }
+
+        return terrain_center
+
+    @property
+    def mesh_resolution(self):
+        """Returns the resolution of the terrain mesh."""
+
+        num_cells = self.mesh.n_cells
+        num_points = self.mesh.n_points
+
+        return num_cells, num_points
+
+    def plot(self):
+        """Returns a plot of the terrain."""
+
+        plot = pv.Plotter()
+        plot.add_mesh(self.mesh)
+        plot.show()
+        plot.close()
+
+
+class WindTerrain(inductiva.scenarios.Scenario):
+    """Physical scenario of a configurable wind terrain simulation.
+
+    This scenario solves steady-state continuity and momentum equations
+    (time-independent) with incompressible flow. 
+    The simulation solves the time-independent equations for several
+    time steps, based on the state of the previous one. The end goal is
+    to determine the steady-state of the system, i.e., where the flow
+    does not change in time anymore.
+    """
+
+    valid_simulators = [fluids.simulators.OpenFOAM]
+
+    def __init__(self,
+                 terrain: Terrain,
+                 wind_velocity: typing.List[float],
+                 wind_position: typing.Optional[typing.List[float]] = None,
+                 altitude_m: float = 300):
+        """Initializes the `WindTerrain` conditions.
+
+        Args:
+            wind_velocity (List): Velocity of the air flow (m/s).
+            wind_position (List): Position of the wind flow (m) above the height
+            of the terrain.
+            terrain (Terrain): Terrain object with the profile of the terrain.
+            altitude_m (float): Altitude above the terrain (m) to establish an
```

Comment:
maybe give it other name to be more explicit of what this is, like atmosphere boundary.

Reply:
But it does not define the boundary... This together with the maximum height establish the top boundary. I will try to rename it.